fv3_cap.F90

```diff
@@ -0,0 +1,1691 @@
+    type(ESMF_Time)                        :: alarm_output_hf_ring, alarm_output_ring
+    type(ESMF_Time)                        :: alarm_output_hf_stop, alarm_output_stop
+    type(ESMF_TimeInterval)                :: RunDuration, timeStep, rsthour, IAU_offsetTI
+    type(ESMF_Config)                      :: cf
+    type(ESMF_RegridMethod_Flag)           :: regridmethod
+    type(ESMF_TimeInterval)                :: earthStep
+    integer(ESMF_KIND_I4)                  :: nhf, nrg
+
+    integer,dimension(6)                   :: date, date_init
+    integer                                :: mpi_comm_atm
+    integer                                :: i, j, k, io_unit, urc, ierr
+    integer                                :: petcount, mype
+    integer                                :: num_output_file
+    logical                                :: OPENED
+    character(ESMF_MAXSTR)                 :: name
+    logical                                :: fcstpe
+    logical,dimension(:), allocatable      :: wrtpe
+    integer,dimension(:), allocatable      :: petList, originPetList, targetPetList
+    character(20)                          :: cwrtcomp
+    character(160)                         :: msg
+    integer                                :: isrcTermProcessing
+
+    character(len=*),parameter  :: subname='(fv3_cap:InitializeAdvertise)'
+    integer nfmout, nfsout , nfmout_hf, nfsout_hf
+    real(kind=8) :: MPI_Wtime, timewri, timeis,timeie,timerhs, timerhe
+!
+!------------------------------------------------------------------------
+!
+    rc = ESMF_SUCCESS
+    timeis = mpi_wtime()
+
+    call ESMF_GridCompGet(gcomp,name=name,vm=vm,rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+    call ESMF_VMGet(vm, mpiCommunicator=mpi_comm_atm,petCount=petcount, &
+                    localpet = mype,rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+    fcstmype = mype
+!    print *,'in fv3_cap,initAdvertize,name=',trim(name),'mpi_comm=',mpi_comm_atm, &
+!       'petcount=',petcount,'mype=',mype
+!
+! create an instance clock for fv3
+    clock_fv3 = ESMF_ClockCreate(clock, rc=RC)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+!
+!------------------------------------------------------------------------
+! get config variables
+!
+    CF = ESMF_ConfigCreate(rc=RC)
+    CALL ESMF_ConfigLoadFile(config=CF ,filename='model_configure' ,rc=RC)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+!
+    num_restart_interval = ESMF_ConfigGetLen(config=CF, label ='restart_interval:',rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+    if(mype == 0) print *,'af nems config,num_restart_interval=',num_restart_interval
```

Comment:
Many print statement to stdout.  Can ESMF log be used?  If stdout prints are needed, perhaps a small subroutine can be introduced to allow for switching these on/off.

Reply:
The ESMF log files are turned off at parallel/operational runs as there are too many log files unless there is a choice to turn on esmf log files on certain tasks. We do need some basic information through the integration